added a prePR command and PR template to remind people to run it (#2583)
typelevel/cats: PULL_REQUEST_TEMPLATE.md

Thank you for contributing to Cats!

> This is a kind reminder that you run `sbt prePR` and commit the changed files before submitting. 

Reviewer: ```suggestion
This is a kind reminder to run `sbt prePR` and commit the changed files before submitting. 
```
